code/modules/scavstation_shared_content/items/egg_yinglet.dm

```diff
@@ -0,0 +1,86 @@
+//yinglet nests and yinglet eggs
+
+//nest
+/obj/structure/yinglet_nest
+	name = "eggbed"
+	icon = 'code/modules/scavstation_shared_content/icons/egg_yinglet.dmi'
+	icon_state = "base"
+	w_class = ITEM_SIZE_HUGE
+	var/has_egg = TRUE
+
+/obj/structure/yinglet_nest/empty
+	has_egg = FALSE
+
+/obj/structure/yinglet_nest/Initialize()
+	. = ..()
+	if(has_egg)
+		new /obj/item/yinglet_egg(src)
+	update_icon()
+	update_desc()
+
+/obj/structure/yinglet_nest/proc/update_desc()
+	desc = "A pile of plain but very soft fabric arranged to carefully hold an egg, as well as prevent the newly hatched younglet from smashing their face when they flop out."
+	if(has_egg && contents.len)
+		for(var/obj/item/I in contents)
+			desc += " There's \a [I] sitting in it."
+
+/obj/structure/yinglet_nest/attack_hand(mob/user)
+	. = ..()
+	if(has_egg && user.a_intent == I_HELP)
+		for(var/obj/item/I in contents)
+			src.visible_message(SPAN_NOTICE("\The [user] pets \the [I]. Grow up big and strong now, okay?"))
+			user.setClickCooldown(1.5 SECONDS)
+		return
+	if(has_egg && user.a_intent != I_HELP)
+		remove_egg(user)
+		return
+
+/obj/structure/yinglet_nest/attackby(obj/item/object, mob/user)
+	. = ..()
+	if(object.w_class > ITEM_SIZE_NORMAL || istype(object, /obj/item/holder))
+		to_chat(user, SPAN_WARNING("\The [object.name] is too big for the [name]!"))
+		return
+	else if(has_egg && contents.len)
+		for(var/obj/item/I in contents)
+			to_chat(user, SPAN_WARNING("\The [name] already has \a [I] in it!"))
+		return
+	else if(!has_egg && object.w_class <= ITEM_SIZE_NORMAL)
+		add_egg(object, user)
+		return
+
+/obj/structure/yinglet_nest/proc/remove_egg(mob/user)
+	has_egg = FALSE
+	for(var/obj/item/E in contents)
+		user.put_in_hands(E)
+		src.visible_message(SPAN_WARNING("\The [user] takes \the [E] from the eggbed."))
+	update_icon()
+	update_desc()
+
+/obj/structure/yinglet_nest/proc/add_egg(var/obj/item/egg, mob/user)
+	has_egg = TRUE
+	user.drop_from_inventory(egg)
+	egg.loc = src
+	src.visible_message(SPAN_NOTICE("\The [user] puts \the [egg] in the eggbed."))
+	update_icon()
+	update_desc()
+
+/obj/structure/yinglet_nest/on_update_icon()
+	. = ..()
+	overlays.Cut()
+	if(contents.len)
+		for(var/obj/item/I in contents)
+			overlays += image("icon" = I.icon, "icon_state" = I.icon_state, "layer" = STRUCTURE_LAYER, "pixel_x" = 0, "pixel_y" = 0)
```

Comment:
You don't need to set `pixel_x` or `pixel_y` if they're going to be 0.
Out of curiosity, does this definitely work? I haven't seen quotation marks used around non-positional arguments before.

Reply:
Delivered this cargo code from trays. It worked there against all reason. I couldn't figure out why so I decided not to question it because I got lazy at this point.

Reply:
nice